### PR TITLE
Proposal: Optionally expose node locality (region, zone, host) to pod

### PR DESCRIPTION
**Pull request by scottcrossen**
Discussion: Has this been investigated before? I imagine that most people want these locality tags in multi-region setups. This PR adds them from [well-known node labels](https://kubernetes.io/docs/reference/labels-annotations-taints/). I'm happy to transfer ownership of my proof-of-concept `initContainer` that [mounts](https://github.com/scottcrossen/kube-node-labels) the node labels in the statefulset.

TLDR: This pullrequest automatically adds the CLI start flag values
```
--locality=region=us-west-2,zone=us-west-2a,host=myhost
```

### CODE REVIEW

**Comment by scottcrossen** [comment]
Rebased

**Comment by scottcrossen** [comment]
Rebased

File: cockroachdb/templates/job.init.yaml [hunk 1]
```diff
@@ -16,9 +16,9 @@ metadata:
   {{- with .Values.labels }}
     {{- toYaml . | nindent 4 }}
   {{- end }}
-  annotations:
```
Comment (sergeyshaykhullin) [inline cockroachdb/templates/job.init.yaml:19]: I don't like this approach, helm hooks annotations should always be present or someone who wants to add new annotations will override helms one

Comment (sergeyshaykhullin) [reply cockroachdb/templates/job.init.yaml:19]: You can add something like extraJobAnnotations and list after helm hooks

Comment (scottcrossen) [reply cockroachdb/templates/job.init.yaml]: Hi! The purpose of this was to make it possible to remove the annotations.

I've moved this discussion into a new PR: https://github.com/cockroachdb/helm-charts/pull/206

**Comment by scottcrossen** [comment]
Rebased PR

File: cockroachdb/templates/job.init.yaml (hunk 1 above)
Comment (sergeyshaykhullin) [reply cockroachdb/templates/job.init.yaml]: Got it. Thanks!
